projects/rero/ng-core/src/lib/record/detail/detail.component.html

```diff
@@ -19,16 +19,28 @@
     <div class="col-9">
         <ng-template ngCoreRecordDetail></ng-template>
     </div>
-    <div class="col-3" *ngIf="record">
-        <a href="javascript:void(0);" class="btn btn-sm btn-secondary" routerLink="../../edit/{{ record.metadata.pid }}">
+    <div class="col-3 text-right" *ngIf="record">
+        <a href class="btn btn-sm btn-primary"
+            routerLink="../../edit/{{ record.metadata.pid }}" *ngIf="canUpdateResult">
             <i class="fa fa-pencil"></i>
             {{ 'Edit' | translate }}
         </a>
-        <a href="javascript:void(0);" class="btn btn-sm btn-secondary ml-1">
-            <i class="fa fa-trash"></i>
-            {{ 'Delete' | translate }}
-        </a>
+        <ng-container *ngIf="canDeleteResult">
+            <a href class="btn btn-sm btn-primary ml-1" [title]="'Delete'|translate"
+                (click)="deleteRecord($event, record.metadata.pid)" *ngIf="canDeleteResult.can; else deleteMessageLink">
+                <i class="fa fa-trash"></i>
+                {{ 'Delete' | translate }}
+            </a>
+            <ng-template #deleteMessageLink>
+                <a href class="btn btn-sm btn-dark ml-1" [title]="'Delete'|translate"
+                    (click)="showDeleteMessage($event, canDeleteResult.message)" *ngIf="canDeleteResult.message">
+                    <i class="fa fa-trash"></i>
+                    {{ 'Delete' | translate }}
+                </a>
+            </ng-template>
+        </ng-container>
+
     </div>
 </div>
 <hr class="my-4">
-<a href="#" (click)="goBack($event)">&laquo; {{ 'Back' | translate }}</a>
+<a href="#" (click)="goBack($event)">&laquo; {{ 'Back' | translate }}</a>
```

Comment:
instead of `href` you can use `routerLink=""` or `type="button"` or use `button`

Reply:
I converted link to button when there's no link to follow, but an action on click